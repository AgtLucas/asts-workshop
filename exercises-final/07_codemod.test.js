import * as babel from 'babel-core'
import * as recast from 'recast'
import {stripIndent} from 'common-tags'
import jQueryAddClassPlugin from './07_codemod'

test('codemods jquery add class', () => {
  const source = stripIndent`
    $(el).addClass(className)
    foo.addClass(otherClassThing)
  `
  const {code} = babel.transform(source, {
    parserOpts: {parser: recast.parse},
    generatorOpts: {generator: recast.print},
    babelrc: false,
    plugins: [jQueryAddClassPlugin],
  })
  expect(code).toMatchSnapshot()
})

test('I submitted my elaboration and feedback', () => {
  const submitted = true
  expect(true).toBe(submitted)
})

//////// EXTRA CREDIT ////////

// If you get this far, try adding a few more tests,
// then file a pull request to add them to the extra credit!
// Learn more here: http://kcd.im/asts-workshop-contributing
