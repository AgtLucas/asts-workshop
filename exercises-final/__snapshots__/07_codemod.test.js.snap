// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codemods jquery add class 1`] = `
"el.classList.add(className)
foo.addClass(otherClassThing)"
`;
